File: backend/app/schemas/portfolio.py
from pydantic import BaseModel, Field
from typing import List, Optional, Dict, Any
from datetime import datetime

class ContactInfo(BaseModel):
    email: Optional[str] = ""
    phone: Optional[str] = ""
    location: Optional[str] = ""
    github: Optional[str] = ""
    linkedin: Optional[str] = ""
    website: Optional[str] = ""

class ExperienceItem(BaseModel):
    id: Optional[str] = ""
    title: str
    company: str
    location: Optional[str] = ""
    period: str
    description: List[str] = Field(default_factory=list)
    technologies: List[str] = Field(default_factory=list)

class ProjectItem(BaseModel):
    id: Optional[str] = ""
    title: str
    description: str
    highlights: List[str] = Field(default_factory=list)
    technologies: List[str] = Field(default_factory=list)
    github_url: Optional[str] = ""
    live_url: Optional[str] = ""
    featured: Optional[bool] = True

class EducationItem(BaseModel):
    id: Optional[str] = ""
    degree: str
    institution: str
    location: Optional[str] = ""
    year: str
    details: Optional[str] = ""

class CertificationItem(BaseModel):
    id: Optional[str] = ""
    name: str
    issuer: str
    year: Optional[str] = ""
    credential_url: Optional[str] = ""

class SkillCategory(BaseModel):
    category: str
    skills: List[str] = Field(default_factory=list)

class PortfolioData(BaseModel):
    name: str
    title: str = "Software Engineer & Innovator"
    about: str
    contact: ContactInfo = Field(default_factory=ContactInfo)
    skills: List[SkillCategory] = Field(default_factory=list)
    experience: List[ExperienceItem] = Field(default_factory=list)
    projects: List[ProjectItem] = Field(default_factory=list)
    education: List[EducationItem] = Field(default_factory=list)
    certifications: List[CertificationItem] = Field(default_factory=list)
    hero_headline: Optional[str] = "Building scalable, elegant digital solutions"
    hero_subheadline: Optional[str] = ""

class Customizations(BaseModel):
    primaryColor: Optional[str] = "#3b82f6"
    fontFamily: Optional[str] = "Inter"
    showCertifications: Optional[bool] = True
    showEducation: Optional[bool] = True
    sectionOrder: List[str] = Field(
        default_factory=lambda: [
            "about",
            "skills",
            "experience",
            "projects",
            "education",
            "certifications",
            "contact"
        ]
    )

class PortfolioCreate(BaseModel):
    title: Optional[str] = "My Portfolio"
    theme: Optional[str] = "modern-glass"
    data: PortfolioData
    customizations: Optional[Customizations] = Field(default_factory=Customizations)

class PortfolioUpdate(BaseModel):
    title: Optional[str] = None
    theme: Optional[str] = None
    data: Optional[PortfolioData] = None
    customizations: Optional[Customizations] = None

class PortfolioResponse(BaseModel):
    id: str
    user_id: Optional[int] = None
    title: str
    theme: str
    data: PortfolioData
    customizations: Optional[Dict[str, Any]] = None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

class ResumeUploadResponse(BaseModel):
    raw_text: str
    portfolio_data: PortfolioData
    ai_enhanced: bool = True

File: frontend/types/portfolio.ts
export interface ContactInfo {
  email?: string;
  phone?: string;
  location?: string;
  github?: string;
  linkedin?: string;
  website?: string;
}

export interface SkillCategory {
  category: string;
  skills: string[];
}

export interface ExperienceItem {
  id?: string;
  title: string;
  company: string;
  location?: string;
  period: string;
  description: string[];
  technologies?: string[];
}

export interface ProjectItem {
  id?: string;
  title: string;
  description: string;
  highlights?: string[];
  technologies?: string[];
  github_url?: string;
  live_url?: string;
  featured?: boolean;
  metrics?: string;
}

export interface EducationItem {
  id?: string;
  degree: string;
  institution: string;
  location?: string;
  year: string;
  details?: string;
}

export interface CertificationItem {
  id?: string;
  name: string;
  issuer: string;
  year?: string;
  credential_url?: string;
}

export interface PortfolioData {
  name: string;
  title: string;
  about: string;
  contact: ContactInfo;
  skills: SkillCategory[];
  experience: ExperienceItem[];
  projects: ProjectItem[];
  education: EducationItem[];
  certifications: CertificationItem[];
  hero_headline?: string;
  hero_subheadline?: string;
}

export interface Customizations {
  primaryColor?: string;
  fontFamily?: string;
  showCertifications?: boolean;
  showEducation?: boolean;
  sectionOrder?: string[];
}

export type ThemeType = 
  | 'modern-glass' 
  | 'executive-slate' 
  | 'cyberpunk-tech' 
  | 'minimal-elegance' 
  | 'dark-prism'
  | 'cyber-security'
  | 'data-scientist';

export interface Portfolio {
  id: string;
  user_id?: number;
  title: string;
  theme: ThemeType;
  data: PortfolioData;
  customizations?: Customizations;
  created_at?: string;
  updated_at?: string;
}

export interface ResumeUploadResponse {
  raw_text: string;
  portfolio_data: PortfolioData;
  ai_enhanced: boolean;
}
